packages/ez_search/ez_search.0.2.1/opam

```diff
@@ -0,0 +1,52 @@
+opam-version: "2.0"
+license: "LGPL-2.1-only with OCaml-LGPL-linking-exception"
+synopsis: "A set of two complementary packages for fulltext searches"
+description: """\
+ocp-search is a tool to grep a pre-indexed set of sources.
+ez_search is a library to search a pre-indexed set of files.
+"""
+authors: ["Fabrice Le Fessant <fabrice.le_fessant@origin-labs.com>"]
+maintainer: ["Fabrice Le Fessant <fabrice.le_fessant@origin-labs.com>"]
+homepage: "https://ocamlpro.github.io/ez_search"
+doc: "https://ocamlpro.github.io/ez_search/sphinx"
+bug-reports: "https://github.com/ocamlpro/ez_search/issues"
+dev-repo: "git+https://github.com/ocamlpro/ez_search.git"
+tags: "org:ocamlpro"
+build: [
+  ["dune" "subst"] {dev}
+  ["sh" "-c" "./scripts/before.sh build '%{name}%'"]
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+  ["sh" "-c" "./scripts/after.sh build '%{name}%'"]
+]
+install: [
+  ["sh" "-c" "./scripts/before.sh install '%{name}%'"]
+]
+depends: [
+  "ocaml" {>= "4.07.0"}
+  "dune" {>= "2.7.0"}
+  "re" {>= "1.8.0" & < "2.0.0"}
+  "parmap" {>= "1.2.3" & < "2.0.0"}
+  "ez_file" {>= "0.2"}
+  "ez_cmdliner" {>= "0.2"}
+  "ppx_inline_test" {with-test}
+  "ppx_expect" {with-test}
+  "odoc" {with-doc}
+  "ocamlformat" {with-test}
+]
+conflicts: [ "base-domains" ]
+available: os ="linux"
```

Comment:
```suggestion
available: os = "linux" & arch != "arm32" & arch != "arm64" & arch != "ppc64" & arch != "s390x"
```

Reply:
This is to address the following failures on arm and s390x:
```
#=== ERROR while compiling ez_search.0.2.1 ====================================#
# context              2.1.1 | linux/arm32 | ocaml-base-compiler.4.13.1 | pinned(https://github.com/ocamlpro/ez_search/archive/v0.2.1.tar.gz)
# path                 ~/.opam/4.13/.opam-switch/build/ez_search.0.2.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ez_search -j 159 @install
# exit-code            1
# env-file             ~/.opam/log/ez_search-2590-ccf2c0.env
# output-file          ~/.opam/log/ez_search-2590-ccf2c0.out
### output ###
#        ocaml src/ez_search/version.ml
# fatal: not a git repository (or any parent up to mount point /)
# Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
# fatal: not a git repository (or any parent up to mount point /)
# Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
#          gcc src/ez_search/strfind.o (exit 1)
# (cd _build/default/src/ez_search && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -O2 -fPIC -DPIC -g -I /home/opam/.opam/4.13/lib/ocaml -I /home/opam/.opam/4.13/lib/cmdliner -I /home/opam/.opam/4.13/lib/ez_cmdliner -I /home/opam/.opam/4.13/lib/ez_file -I /home/opam/.opam/4.13/lib/ez_subst -I /home/opam/.opam/4.13/lib/ocplib_stuff -I /home/opam/.opam/4.13/lib/parmap -I /home/opam/.opam/4.13/lib/re -I /home/opam/.opam/4.13/lib/seq -o strfind.o -c strfind.c)
# strfind.c:12:10: fatal error: x86intrin.h: No such file or directory
#    12 | #include <x86intrin.h>
#       |          ^~~~~~~~~~~~~
# compilation terminated.
```

And the following on PPC64:
```
#=== ERROR while compiling ez_search.0.2.1 ====================================#
# context              2.1.1 | linux/ppc64 | ocaml-base-compiler.4.13.1 | pinned(https://github.com/ocamlpro/ez_search/archive/v0.2.1.tar.gz)
# path                 ~/.opam/4.13/.opam-switch/build/ez_search.0.2.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ez_search -j 79 @install
# exit-code            1
# env-file             ~/.opam/log/ez_search-3099-202cc3.env
# output-file          ~/.opam/log/ez_search-3099-202cc3.out
### output ###
#        ocaml src/ez_search/version.ml
# fatal: not a git repository (or any parent up to mount point /)
# Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
# fatal: not a git repository (or any parent up to mount point /)
# Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
#          gcc src/ez_search/strfind.o (exit 1)
# (cd _build/default/src/ez_search && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -O2 -fPIC -DPIC -g -I /home/opam/.opam/4.13/lib/ocaml -I /home/opam/.opam/4.13/lib/cmdliner -I /home/opam/.opam/4.13/lib/ez_cmdliner -I /home/opam/.opam/4.13/lib/ez_file -I /home/opam/.opam/4.13/lib/ez_subst -I /home/opam/.opam/4.13/lib/ocplib_stuff -I /home/opam/.opam/4.13/lib/parmap -I /home/opam/.opam/4.13/lib/re -I /home/opam/.opam/4.13/lib/seq -o strfind.o -c strfind.c)
# In file included from strfind.c:12:
# /usr/lib/gcc/powerpc64le-linux-gnu/10/include/x86intrin.h:32:2: error: #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#    32 | #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#       |  ^~~~~
# In file included from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/x86intrin.h:39,
#                  from strfind.c:12:
# /usr/lib/gcc/powerpc64le-linux-gnu/10/include/mmintrin.h:52:2: error: #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#    52 | #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#       |  ^~~~~
# In file included from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/x86intrin.h:41,
#                  from strfind.c:12:
# /usr/lib/gcc/powerpc64le-linux-gnu/10/include/xmmintrin.h:54:2: error: #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#    54 | #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#       |  ^~~~~
# In file included from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/xmmintrin.h:79,
#                  from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/x86intrin.h:41,
#                  from strfind.c:12:
# /usr/lib/gcc/powerpc64le-linux-gnu/10/include/mmintrin.h:52:2: error: #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#    52 | #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#       |  ^~~~~
# In file included from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/x86intrin.h:43,
#                  from strfind.c:12:
# /usr/lib/gcc/powerpc64le-linux-gnu/10/include/emmintrin.h:56:2: error: #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#    56 | #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#       |  ^~~~~
# In file included from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/emmintrin.h:66,
#                  from /usr/lib/gcc/powerpc64le-linux-gnu/10/include/x86intrin.h:43,
#                  from strfind.c:12:
# /usr/lib/gcc/powerpc64le-linux-gnu/10/include/xmmintrin.h:54:2: error: #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#    54 | #error "Please read comment above.  Use -DNO_WARN_X86_INTRINSICS to disable this error."
#       |  ^~~~~
# strfind.c:13:9: error: #pragma GCC target 'avx2' is invalid
#    13 | #pragma GCC target("avx2")
#       |         ^~~
# strfind.c:38:1: error: unknown type name '__m256i'
#    38 | __m256i hsum_epu8_epu64(__m256i v) {
#       | ^~~~~~~
# strfind.c:38:25: error: unknown type name '__m256i'
#    38 | __m256i hsum_epu8_epu64(__m256i v) {
#       |                         ^~~~~~~
# strfind.c:43:28: error: unknown type name '__m256i'
#    43 | uint64_t hsum_epu64_scalar(__m256i v) {
#       |                            ^~~~~~~
# strfind.c:53:8: error: unknown type name '__m256i'
#    53 | static __m256i count_one_pack(const char *needle, int needle_len,
#       |        ^~~~~~~
# strfind.c: In function 'count_one_pack':
# strfind.c:56:3: error: unknown type name '__m256i'
#    56 |   __m256i mask = _mm256_set1_epi8 (0xFF);
#       |   ^~~~~~~
# strfind.c:56:18: warning: implicit declaration of function '_mm256_set1_epi8'; did you mean '_mm_set1_epi8'? [-Wimplicit-function-declaration]
#    56 |   __m256i mask = _mm256_set1_epi8 (0xFF);
#       |                  ^~~~~~~~~~~~~~~~
#       |                  _mm_set1_epi8
# strfind.c:58:5: error: unknown type name '__m256i'
#    58 |     __m256i tester = _mm256_set1_epi8 (needle[i]);
#       |     ^~~~~~~
# strfind.c:59:5: error: unknown type name '__m256i'
#    59 |     __m256i v = _mm256_lddqu_si256((__m256i const *)(haystack + pos + i));
#       |     ^~~~~~~
# strfind.c:59:17: warning: implicit declaration of function '_mm256_lddqu_si256' [-Wimplicit-function-declaration]
#    59 |     __m256i v = _mm256_lddqu_si256((__m256i const *)(haystack + pos + i));
#       |                 ^~~~~~~~~~~~~~~~~~
# strfind.c:59:37: error: '__m256i' undeclared (first use in this function)
#    59 |     __m256i v = _mm256_lddqu_si256((__m256i const *)(haystack + pos + i));
#       |                                     ^~~~~~~
# strfind.c:59:37: note: each undeclared identifier is reported only once for each function it appears in
# strfind.c:59:44: error: expected ')' before 'const'
#    59 |     __m256i v = _mm256_lddqu_si256((__m256i const *)(haystack + pos + i));
#       |                                    ~       ^~~~~~
#       |                                            )
# strfind.c:60:12: error: expected ';' before 'eq_char'
#    60 |     __m256i eq_char = _mm256_cmpeq_epi8 (v, tester);
#       |            ^~~~~~~~
#       |            ;
# strfind.c:61:12: warning: implicit declaration of function '_mm256_and_si256'; did you mean '_mm_and_si128'? [-Wimplicit-function-declaration]
#    61 |     mask = _mm256_and_si256 (mask, eq_char);
#       |            ^~~~~~~~~~~~~~~~
#       |            _mm_and_si128
# strfind.c:61:36: error: 'eq_char' undeclared (first use in this function); did you mean 'u_char'?
#    61 |     mask = _mm256_and_si256 (mask, eq_char);
#       |                                    ^~~~~~~
#       |                                    u_char
# strfind.c:62:8: warning: implicit declaration of function '_mm256_testz_si256' [-Wimplicit-function-declaration]
#    62 |     if(_mm256_testz_si256(mask, mask))
#       |        ^~~~~~~~~~~~~~~~~~
# strfind.c:63:14: warning: implicit declaration of function '_mm256_setzero_si256'; did you mean '_mm_setzero_si128'? [-Wimplicit-function-declaration]
#    63 |       return _mm256_setzero_si256();
#       |              ^~~~~~~~~~~~~~~~~~~~
#       |              _mm_setzero_si128
# strfind.c:65:10: error: expected ';' before 'ones'
#    65 |   __m256i ones = _mm256_sub_epi8(_mm256_setzero_si256(),mask);
#       |          ^~~~~
#       |          ;
# strfind.c:66:10: error: expected ';' before 'sum_64'
#    66 |   __m256i sum_64 = hsum_epu8_epu64(ones);
#       |          ^~~~~~~
#       |          ;
# strfind.c:67:10: error: 'sum_64' undeclared (first use in this function)
#    67 |   return sum_64;
#       |          ^~~~~~
# strfind.c: In function 'count_matches':
# strfind.c:73:3: error: unknown type name '__m256i'
#    73 |   __m256i count_64 = _mm256_setzero_si256();
#       |   ^~~~~~~
# strfind.c:76:16: warning: implicit declaration of function '_mm256_add_epi64'; did you mean '_mm_add_epi64'? [-Wimplicit-function-declaration]
#    76 |     count_64 = _mm256_add_epi64(count_64, count_one_pack(needle, needle_len, haystack, i * 32));
#       |                ^~~~~~~~~~~~~~~~
#       |                _mm_add_epi64
# strfind.c:78:20: warning: implicit declaration of function 'hsum_epu64_scalar' [-Wimplicit-function-declaration]
#    78 |   uint64_t count = hsum_epu64_scalar(count_64);
#       |                    ^~~~~~~~~~~~~~~~~
```